openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -0,0 +1,479 @@
+                    AND imd.res_id = armt.id)
+            WHERE imd.id IS NULL) armt2
+        WHERE has_second_line AND (rule_type != 'invoice_matching' OR (
+            rule_type = 'invoice_matching' AND match_total_amount
+            AND match_total_amount_param < 100.0))
+        ORDER BY id""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_reconcile_model_line_template_account_tax_template_rel (
+            account_reconcile_model_line_template_id, account_tax_template_id)
+        SELECT armlt.id, rel.account_tax_template_id
+        FROM account_reconcile_model_template_account_tax_template_rel rel
+        JOIN account_reconcile_model_line_template armlt
+            ON rel.account_reconcile_model_template_id = armlt.model_id""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_reconcile_model_line_template_account_tax_template_rel (
+            account_reconcile_model_line_template_id, account_tax_template_id)
+        SELECT armlt.id, rel.account_tax_template_id
+        FROM account_reconcile_model_tmpl_account_tax_bis_rel rel
+        JOIN account_reconcile_model_line_template armlt
+            ON rel.account_reconcile_model_template_id = armlt.model_id""",
+    )
+
+
+def create_account_tax_report_lines(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_tax_report
+        ADD COLUMN root_line_id integer""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_tax_report (name, country_id, root_line_id,
+            create_uid, create_date, write_uid, write_date)
+        SELECT name, country_id, id, create_uid, create_date, write_uid, write_date
+        FROM account_tax_report_line
+        WHERE parent_id IS NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_tax_report_line atrl
+        SET report_id = atr.id
+        FROM account_tax_report atr
+        WHERE atr.root_line_id = atrl.id
+        """,
+    )
+    while True:
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE account_tax_report_line atrl
+            SET report_id = atrl2.report_id
+            FROM account_tax_report_line atrl2
+            WHERE atrl.parent_id = atrl2.id AND atrl.report_id IS NULL
+            RETURNING atrl.id""",
+        )
+        if not env.cr.fetchone():
+            break
+
+
+def post_statements_with_unreconciled_lines(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_bank_statement bst
+        SET state = 'posted'
+        FROM account_bank_statement_line bstl
+        WHERE bst.state = 'confirm' AND bstl.statement_id = bst.id
+            AND bstl.is_reconciled IS DISTINCT FROM TRUE
+        """,
+    )
+
+
+def pass_bank_statement_line_note_to_invoice_narration(env):
```

Comment:
I believe the related `account.move`s are 'journal entry' move, not invoice moves (damn you account-pocalypse) so the method names here and the one down below are slightly misleading.

Reply:
methods renamed hehe